pkg/helm/run.go

```diff
@@ -125,7 +125,7 @@ func Run(flags *hoflags.HelmOperatorFlags) error {
 			GVK:                     w.GroupVersionKind,
 			ManagerFactory:          release.NewManagerFactory(mgr, w.ChartDir),
 			ReconcilePeriod:         flags.ReconcilePeriod,
-			WatchDependentResources: w.WatchDependentResources,
+			WatchDependentResources: *w.WatchDependentResources,
```

Comment:
Any reason why this needs to be a pointer?

Reply:
This was to make `yaml.Marshal` respect the difference between this field being present or not present in the YAML file.

Reply:
As I understand, while unmarshal and marshal, this will make `yaml: "omitempty"` effective.